TTS/speaker_encoder/dataset.py

```diff
@@ -90,28 +104,21 @@ def __parse_items(self):
 
         self.speakers = [k for (k, v) in self.speaker_to_utters.items()]
 
-    # def __parse_items(self):
-    #     """
-    #     Find unique speaker ids and create a dict mapping utterances from speaker id
-    #     """
-    #     speakers = list({item[-1] for item in self.items})
-    #     self.speaker_to_utters = {}
-    #     self.speakers = []
-    #     for speaker in speakers:
-    #         speaker_utters = [item[1] for item in self.items if item[2] == speaker]
-    #         if len(speaker_utters) < self.num_utter_per_speaker and self.skip_speakers:
-    #             print(
-    #                 f" [!] Skipped speaker {speaker}. Not enough utterances {self.num_utter_per_speaker} vs {len(speaker_utters)}."
-    #             )
-    #         else:
-    #             self.speakers.append(speaker)
-    #             self.speaker_to_utters[speaker] = speaker_utters
-
     def __len__(self):
         return int(1e10)
 
-    def __sample_speaker(self):
+    def get_num_speakers(self):
+        return len(self.speakers)
+
+    def __sample_speaker(self, ignore_speakers=None):
         speaker = random.sample(self.speakers, 1)[0]
+        # if list of speakers_id is provide make sure that it's will be ignored
+        if ignore_speakers and self.speakerid_to_classid[speaker] in ignore_speakers:
+            while True:
+                speaker = random.sample(self.speakers, 1)[0]
+                if self.speakerid_to_classid[speaker] not in ignore_speakers:
+                    break
```

Comment:
I don't think so, SpeakerManager is to control speaker.json and this is not directly connected. In addition, I think it will not be useful elsewhere.

Reply:
I am changing the SpeakerManager since it was my intention to replace all the speaker-related functions with the speaker manager. So I think it is better to keep your PR on hold until I pull the new `SpealerManager`. Then we can add it here.